tests/components/dynalite/test_config_flow.py

```diff
@@ -1,36 +1,90 @@
 """Test Dynalite config flow."""
-from unittest.mock import Mock, call, patch
-
-from homeassistant.components.dynalite import config_flow
-
-from tests.common import mock_coro
-
-
-async def test_step_import():
-    """Test a successful setup."""
-    flow_handler = config_flow.DynaliteFlowHandler()
-    with patch.object(flow_handler, "context", create=True):
-        with patch.object(flow_handler, "hass", create=True) as mock_hass:
-            with patch.object(
-                flow_handler, "async_create_entry", create=True
-            ) as mock_create:
-                host = "1.2.3.4"
-                entry1 = Mock()
-                entry1.data = {"host": host}
-                entry2 = Mock()
-                entry2.data = {"host": "5.5"}
-                mock_hass.config_entries.async_entries = Mock(
-                    return_value=[entry1, entry2]
-                )
-                mock_hass.config_entries.async_remove = Mock(
-                    return_value=mock_coro(Mock())
-                )
-                await flow_handler.async_step_import({"host": "1.2.3.4"})
-                mock_hass.config_entries.async_remove.assert_called_once()
-                assert mock_hass.config_entries.async_remove.mock_calls[0] == call(
-                    entry1.entry_id
-                )
-                mock_create.assert_called_once()
-                assert mock_create.mock_calls[0] == call(
-                    title=host, data={"host": host}
-                )
+from asynctest import patch
+
+from homeassistant import config_entries
+from homeassistant.components import dynalite
+
+from tests.common import MockConfigEntry
+
+
+async def run_flow(hass, setup, connection):
+    """Run a flow with or without errors and return result."""
+    host = "1.2.3.4"
+    with patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.async_setup",
+        return_value=setup,
+    ), patch(
+        "homeassistant.components.dynalite.bridge.DynaliteDevices.available", connection
+    ), patch(
+        "homeassistant.components.dynalite.bridge.CONNECT_INTERVAL", 0
+    ):
+        result = await hass.config_entries.flow.async_init(
+            dynalite.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={dynalite.CONF_HOST: host},
+        )
+    return result
+
+
+async def test_flow_works(hass):
+    """Test a successful config flow."""
+    result = await run_flow(hass, True, True)
+    assert result["type"] == "create_entry"
+
+
+async def test_flow_setup_fails(hass):
+    """Test a flow where async_setup fails."""
+    result = await run_flow(hass, False, True)
+    assert result["type"] == "abort"
+    assert result["reason"] == "bridge_setup_failed"
+
+
+async def test_flow_no_connection(hass):
+    """Test a flow where connection times out."""
+    result = await run_flow(hass, True, False)
+    assert result["type"] == "abort"
+    assert result["reason"] == "no_connection"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5051600)